### DISCUSSION TITLE
High Memory Comsumption

### DISCUSSION THREAD

**Discussion by rjpowers10** [OrchardCMS/OrchardCore#8683] [Q&A]
Hello all,

I've been struggling with server memory issues for the past few days and I'm looking for advice. I didn't want to open a bug yet since I don't have concrete evidence it's anything that OC is doing anything wrong. It very well could be my own application code that is the problem, but for the most part we're just using this site to display static content items. However, we do have a decent number of content items, around 9,500.

On a fresh start the site uses about 325MB. Seems reasonable. The problem is that is quickly grows to upwards of 3000MB in only a few hours. This isn't a high traffic site, with probably less than a dozen or so users at a given time. The other day I saw it at almost 7000MB and had to restart the site, but since then it hasn't climbed that high. But it does frequently top out between 3000MB and 4000MB.

Just taking a look through the OC code, I do see several uses of `IMemoryCache`. Again, I don't have any evidence this is the problem but one thing that stuck me as odd is that it doesn't seem like an expiration is being set on many of the cache entries. My understanding is that these expiration-less entries won't be evicted unless the `Compact` method is called, but I could be missing something. EDIT: I just realized `Compact` is part of `MemoryCache`, not `IMemoryCache`.

Sorry I don't have more to go on. Figuring out memory issues is a new area for me. Let me know if there's something more I can provide. Thank you!

EDIT: Maybe related to #9060? Also see #7826.

**Comment by Piedone** [comment]
If anyone stumbles upon this: for v1.8, we fixed multiple memory issues, most prominently https://github.com/OrchardCMS/OrchardCore/issues/14117.
